app/auth/identity.py

```
@@ -79,7 +79,7 @@ def __init__(self, obj=None, token=None):
                 raise ValueError("The account_number is mandatory.")
             elif not self.identity_type or self.identity_type not in IdentityType.__members__.values():
                 raise ValueError("Identity type invalid or missing in provided Identity")
-            elif self.auth_type and self.auth_type not in AuthType.__members__.values():
+            elif self.auth_type is not None and self.auth_type not in AuthType.__members__.values():
```

Deleting `is not None` is cleaner

---

Yes, but it changes the functionality. If `self.auth_type` is set to an empty string, it counts as None, which breaks the logic.
If someone doesn't provide `self.auth_type`, we should be OK with it; if they set it to an empty string, we need to throw an error.